packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.internal.js

```diff
@@ -0,0 +1,1427 @@
+          </div>
+        );
+      }
+
+      render(
+        <>
+          <Example1 />
+          <Example2 />
+        </>,
+        container,
+      );
+
+      const matches = findAllNodes(document.body, [
+        createTestNameSelector('match'),
+      ]);
+      expect(matches).toHaveLength(3);
+      expect(matches.map(m => m.id).sort()).toEqual([
+        'match1',
+        'match2',
+        'match3',
+      ]);
+    });
+
+    it('should ignore nested matches', () => {
+      function Example() {
+        return (
+          <div data-testname="match" id="match1">
+            <div data-testname="match" id="match2" />
+          </div>
+        );
+      }
+
+      render(<Example />, container);
+
+      const matches = findAllNodes(document.body, [
+        createComponentSelector(Example),
+        createTestNameSelector('match'),
+      ]);
+      expect(matches).toHaveLength(1);
+      expect(matches[0].id).toEqual('match1');
+    });
+
+    it('should enforce the specific order of selectors', () => {
+      function Outer() {
+        return (
+          <>
+            <div data-testname="match" id="match1" />
+            <Inner />
+          </>
+        );
+      }
+      function Inner() {
+        return <div data-testname="match" id="match1" />;
+      }
+
+      render(<Outer />, container);
+
+      expect(
+        findAllNodes(document.body, [
+          createComponentSelector(Inner),
+          createComponentSelector(Outer),
+          createTestNameSelector('match'),
+        ]),
+      ).toHaveLength(0);
+    });
+
+    it('should not search within hidden subtrees', () => {
+      const ref1 = React.createRef(null);
+      const ref2 = React.createRef(null);
+
+      function Outer() {
+        return (
+          <>
+            <div hidden={true}>
+              <div ref={ref1} data-testname="match" />
+            </div>
+            <Inner />
+          </>
+        );
+      }
+      function Inner() {
+        return <div ref={ref2} data-testname="match" />;
+      }
+
+      render(<Outer />, container);
+
+      const matches = findAllNodes(document.body, [
+        createComponentSelector(Outer),
+        createTestNameSelector('match'),
+      ]);
+
+      expect(matches).toHaveLength(1);
+      expect(matches[0]).toBe(ref2.current);
+    });
+
+    it('should support filtering by display text', () => {
+      function Example() {
+        return (
+          <div>
+            <div>foo</div>
+            <div>
+              <div id="match">bar</div>
+            </div>
+          </div>
+        );
+      }
+
+      render(<Example />, container);
+
+      const matches = findAllNodes(document.body, [
+        createComponentSelector(Example),
+        createTextSelector('bar'),
+      ]);
+      expect(matches).toHaveLength(1);
+      expect(matches[0].id).toBe('match');
+    });
+
+    it('should support filtering by explicit accessibiliy role', () => {
```

Comment:
For `dom-accessibility-role` the first role in the `role` attribute was sufficient but in `@testing-library/dom` we had some requests supporting fallback roles e.g. `<div role="meter progressbar" />`. https://github.com/testing-library/dom-testing-library/issues/411 has some more context in case that is something you're interested in.

Reply:
Cool. Thanks for the pointer. Might as well handle that case too since it's easy enough!